### PR TITLE
nested namepace and related cleanup.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR
No semantic changes

### CODE REVIEW

**Review by havardpe — CHANGES_REQUESTED** [review]
please revert the auto-formatting performed on searchlib/src/vespa/searchlib/features/element_similarity_feature.cpp

**Comment by baldersheim** [comment]
@havardpe Fixed

File: searchlib/src/vespa/searchlib/features/element_similarity_feature.cpp [hunk 1]
```diff
@@ -414,8 +427,7 @@ ElementSimilarityBlueprint::setup(const fef::IIndexEnvironment &env,
 }
 
 fef::FeatureExecutor &
-ElementSimilarityBlueprint::createExecutor(const fef::IQueryEnvironment &env, vespalib::Stash &stash) const
-{
+ElementSimilarityBlueprint::createExecutor(const fef::IQueryEnvironment &env, vespalib::Stash &stash) const {
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/element_similarity_feature.cpp]: class functions should have open-brace on separate line

**Review by havardpe — COMMENTED** [review]
some minor comments

File: searchlib/src/vespa/searchlib/features/dotproductfeature.h [hunk 2]
```diff
@@ -246,7 +242,7 @@ class SparseDotProductByContentFillExecutor : public DotProductExecutorBase<Base
     SparseDotProductByContentFillExecutor(const attribute::IAttributeVector * attribute,
                                           const V & queryVector,
                                           const IV & queryIndexes);
-    ~SparseDotProductByContentFillExecutor();
+    ~SparseDotProductByContentFillExecutor() override;
```
[the remark this answers is not in the document]
Comment (vekterli) [reply searchlib/src/vespa/searchlib/features/dotproductfeature.h]: `override` is preferred on destructors of child types in hierarchies containing virtual functions, as its enforces that the parent has implemented a virtual destructor. I.e. no sneaky resource leaks. Compilers will often warn on this (no-virtual-dtor etc), but good to not have to depend on this.

File: searchlib/src/vespa/searchlib/features/element_similarity_feature.cpp [hunk 3]
```diff
@@ -363,8 +378,7 @@ ElementSimilarityBlueprint::visitDumpFeatures(const fef::IIndexEnvironment &env,
         const fef::FieldInfo &field = *env.getField(i);
         if ((field.type() == fef::FieldType::INDEX) &&
             (field.collection() != CollectionType::SINGLE) &&
-            ( ! field.isFilter()))
-        {
+            (!field.isFilter())) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/element_similarity_feature.cpp]: I do too, fixed

File: searchlib/src/vespa/searchlib/features/dotproductfeature.h [hunk 4]
```diff
@@ -272,7 +267,7 @@ class DotProductBlueprint : public fef::Blueprint {
 
 public:
     DotProductBlueprint();
-    ~DotProductBlueprint();
+    ~DotProductBlueprint() override;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/dotproductfeature.h]: See above.

**Pull request merged**